### PR TITLE
Fixes #149

### LINKED ISSUE #149: Continuing Intro.js for User Onboarding in the Create Note Page
**Original Issue** https://github.com/oss-slu/lrda_website/issues/138

**Description**

**As** a new user,
**I need** an Intro.js tour for the create note page to understand its main features,
**So that** I can easily navigate and use the notes functionalities and refer back to the tour through the help section if needed.

**Tasks**
- [x] add code in notes elements module to start the IntroJs Tour
    - [x] Reference the correct button IDs to ensure the tour is triggered properly.
- [x] Ensure the implementation is regression-free.
- [x] Add a "Skip Tour" button that allows users to exit the tour if they do not wish to continue across multiple pages.
- [x] Create a test cases for your code
    - [x] Ensure that the tour starts correctly.
    - [x] Ensure that users can skip or continue the tour seamlessly.
    - [x] Ensure that the tour progresses step-by-step without issues.
- [x] regression free
    - committed code will not break other functionalities or previous commits

**Acceptance Criteria**
- [x] Create at least one test case for you code
    - [x] Unit test to check functionality.
    - [x] Integration test to verify correct operation across components.
- [x] Code is well commented
- [x] Commit only the changes that you’ve made
- [x] Code should pass the test case before merge
- [x] introJs Triggers tour from create note page
- [x] the tour is Continuing from Map/page
- [x] the ability to skip tour
- [x] the tour need to remember its steps so the user can go back and forth between pages and the tour will resume from its previous step
- [x] Logout button need to be referenced at the end of the tour not in map/page

Code example and test example can be found in branch "Wheres-Religion-Desktop-fall-2024"

This issue will help members new to the project to understand the work flow of the application and familiarize with the project.

### PR DESCRIPTION
**Pull request by amychen108**
**1. PR Reference:** 

fixes #149 
**2. What was changed:**
Continuing the guided tour on the "Create a Note" page on the website. It should be triggered automatically after the intro on the map page finishes and it should be seamless. There will be 5+ popups on the Create a Note page. Also adding a skip button so the user can skip the intro if they'd like.
**3. Why it was changed:**
This change was needed to give the user a smooth tutorial experience which shows each of the features regarding note creation such as adding a title, saving and deleting your note along with other features like attaching videos and images, etc.
**4. How it was changed:**
- Added the intro.js to the note_compoment.tsx where the intro should be occuring after the maps page
- Added intro.js to the side_bar.tsx so the intro clicks the "add note" button and continues the intro on the note_component.tsx
- Changed skipLabel in the intro to "Skip"
- Created each of the intro popups for the title, save, delete, date/time, location.
- Created test cases for each of these popups in the introjs.test.tsx

### CODE REVIEW

File: app/lib/components/side_bar.tsx [hunk 1]
```diff
@@ -78,14 +110,17 @@ const Sidebar: React.FC<SidebarProps> = ({ onNoteSelect }) => {
     <div className="h-[90vh] bg-gray-200 p-4 overflow-y-auto flex flex-col z-30">
       <div className="w-full mb-4">
         <SearchBarNote onSearch={handleSearch} />
-      </div>
-      <Button data-testid="add-note-button" onClick={handleAddNote}>
+
```
Comment (jwang-101) [inline app/lib/components/side_bar.tsx:113]: Maybe fix some spacing stuff here, looks inconsistent

File: app/lib/components/noteElements/note_component.tsx [hunk 2]
```diff
@@ -309,7 +372,9 @@ export default function NoteEditor({
               />
             </div>
             <div className="w-2 h-9 bg-border" />
-            <div className="bg-white p-2 rounded">
+            <div className="bg-white p-2 rounded"
+            ref = {locationRef}
```
Comment (jwang-101) [inline app/lib/components/noteElements/note_component.tsx:376]: Inconsistent formatting (compared to line 366)

**Review by jwang-101 — COMMENTED** [review]
Just some spacing and formatting inconsistencies.

**Review by BaloneyBoy97 — COMMENTED** [review]
Great work overall,
Product Mindset:
- added feature is aligned with the assigned task
- introJs Triggers tour from create note page is in line with client priority
- code changes is Continuing from Map/page is easy to understand and is in line with project architecture

Code Quality:
- the fundamentals of the code is sound
- performance is passable, introjs popup is located in the correct page
- good comments in your code, short and straightforward.
- Still need to create test cases and pass

Functionality verification:
- the code is running with no compile error
- the feature does not yet meet the task requirement.
    - need to finish up unit test
- Code is regression free atm

Focus On The Important Issues:
- Changes are focused on the priority issue

Good job Amy, with your first important issue. Keep up and work and remember to add test cases for your code before Monday to have it ready to merge

**Review by BaloneyBoy97 — APPROVED** [review]
Great work overall,

Good job Amy, with your second sprint issue.

**Review by yashb196 — APPROVED** [review]
Good job amy approving your PR

**Pull request merged**